core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -80,21 +81,26 @@ public static Transaction beginCreate(File temp, String name, Schema schema,
   }
 
   public static Transaction beginReplace(File temp, String name, Schema schema, PartitionSpec spec) {
-    return beginReplace(temp, name, schema, spec, SortOrder.unsorted(), ImmutableMap.of());
+    return beginReplace(temp, name, schema, spec, SortOrder.unsorted(), ImmutableMap.of(), null);
   }
 
   public static Transaction beginReplace(File temp, String name, Schema schema, PartitionSpec spec,
-                                         SortOrder sortOrder, Map<String, String> properties) {
-    TestTableOperations ops = new TestTableOperations(name, temp);
-    TableMetadata current = ops.current();
+      SortOrder sortOrder, Map<String, String> properties) {
+    return beginReplace(temp, name, schema, spec, sortOrder, properties, null);
+  }
+
+  public static Transaction beginReplace(File temp, String name, Schema schema, PartitionSpec spec,
+                                         SortOrder sortOrder, Map<String, String> properties, TestTableOperations ops) {
+    TestTableOperations finalOps = ops != null ? ops : new TestTableOperations(name, temp);
```

Comment:
Instead of the null check here, you can pass the new TestTableOperations() in the previous method.

```java
    return beginReplace(temp, name, schema, spec, sortOrder, properties, new TestTableOperations(name, temp));
```

I tend to like that better because then the caller method knows exactly what value will be used there instead of having to check the implementation for the special null casing. If this was a production method I would put a precondition (ops not null) but I'm not sure that's important here since we know all the callers.

Reply:
sounds like a good practice